dbt/adapters/bigquery.py

```diff
@@ -0,0 +1,300 @@
+            view.create()
+
+        if view.created is None:
+            raise RuntimeError("Error creating view {}".format(model_name))
+
+        return "CREATE VIEW"
+
+    @classmethod
+    def fetch_query_results(cls, query):
+        all_rows = []
+
+        rows = query.rows
+        token = query.page_token
+
+        while True:
+            all_rows.extend(rows)
+            if token is None:
+                break
+            rows, total_count, token = query.fetch_data(page_token=token)
+        return rows
+
+    @classmethod
+    def execute_and_fetch(cls, profile, sql, model_name=None, **kwargs):
+        conn = cls.get_connection(profile, model_name)
+        client = conn.get('handle')
+
+        formatted_sql = cls.format_sql_for_bigquery(sql)
+        query = client.run_sync_query(formatted_sql)
+        query.timeout_ms = cls.QUERY_TIMEOUT
+
+        debug_message = "Fetching data for query {}:\n{}"
+        logger.debug(debug_message.format(model_name, formatted_sql))
+
+        query.run()
+
+        return cls.fetch_query_results(query)
+
+    @classmethod
+    def add_begin_query(cls, profile, name):
+        raise Exception("not implemented")
+        return cls.add_query(profile, 'BEGIN', name, auto_begin=False,
+                             select_schema=False)
+
+    @classmethod
+    def create_schema(cls, profile, schema, model_name=None):
+        logger.debug('Creating schema "%s".', schema)
+
+        dataset = cls.get_dataset(profile, schema, model_name)
+
+        with cls.exception_handler(profile, 'create dataset', model_name):
+            dataset.create()
+
+    @classmethod
+    def check_schema_exists(cls, profile, schema, model_name=None):
+        conn = cls.get_connection(profile, model_name)
+
+        client = conn.get('handle')
+
+        with cls.exception_handler(profile, 'create dataset', model_name):
+            all_datasets = client.list_datasets()
+            return any([ds.name == schema for ds in all_datasets])
+
+    @classmethod
+    def get_dataset(cls, profile, dataset_name, model_name=None):
+        conn = cls.get_connection(profile, model_name)
+
+        client = conn.get('handle')
+        dataset = client.dataset(dataset_name)
+        return dataset
+
+    @classmethod
+    def add_query(cls, profile, sql, model_name=None, auto_begin=True):
+        raise Exception("Not implemented")
+
+    @classmethod
+    def cancel_connection(cls, profile, connection):
+        raise Exception("Not implemented")
+        handle = connection['handle']
+        sid = handle.session_id
+
+        connection_name = connection.get('name')
+
+        sql = 'select system$abort_session({})'.format(sid)
+
+        logger.debug("Cancelling query '{}' ({})".format(connection_name, sid))
+
+        _, cursor = cls.add_query(profile, sql, 'master')
+        res = cursor.fetchone()
+
+        logger.debug("Cancel query '{}': {}".format(connection_name, res))
```

Comment:
@cmcarthur it doesn't look like it's possible to cancel a table creation request. Jobs can be canceled though, which may be good to keep in mind for the future. This branch will log an error that looks like this on ctrl-c:

```
The bigquery adapter does not support query cancellation. Some queries may still be running!
```